src/main/kotlin/org/rust/lang/core/grammar/rust.bnf

```diff
@@ -280,6 +280,8 @@
     generateTokens=false
     generateTokenAccessors=true
 
+    // Preserve information about missing ':'
+    consumeTokenMethod("type_ascription") = "consumeToken"
```

Comment:
As fair as I understand from comparing `consumeToken` and `consumeTokenFast`, this line makes Grammar-Kit push information about missing `':'` to `ErrorState`, when consuming it fails at `type_ascription`. This information is retrieved when error message is generated in users of this production. That's why `':'` popped out in [`struct_pat.txt`](https://github.com/intellij-rust/intellij-rust/pull/337/files#diff-b899c29d69c1bb354a1c4e69ac12b213R44).

Fast consuming doesn't do this and instead quickly exits `type_ascription` parser when there is no `':'` to eat.